### PR TITLE
B:18828 - Add Create Okta Account Logic to UpdateRequestedOfficeUser(MERGE AFTER B-18821 & B-18825)(READY)

### PR DESCRIPTION
**Pull request by TevinAdams**
## Summary

MERGE AFTER [B-18821](https://github.com/transcom/mymove/pull/12281) - MERGED
MERGE AFTER [B-18825](https://github.com/transcom/mymove/pull/12288) - MERGED

This PR introduces additions to the UpdateRequestedOfficeUserHandler. These additions includes the following:
1. Create okta Account Logic - (This only happens in the case that param.Status == "APPROVED")
2. UpdateRequestedOfficeUser will always run in all cases EXCEPT the case when CreateAccountOkta logic runs and fails.

[Okta API Specifications](https://developer.okta.com/docs/reference/api/roles/) explains more about the okta api specifications

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Start the client & server and log in as a MM Admin
2. Using postman or swagger UI
3. Send a PATCH request to /requested-office-users/{officeUserId} (Make sure Status is set to "APPROVED")
4. Provide relevant needed data for Okta account creation such as otherUniqueId, Email, Firstname, Lastname, gsaID, EDIPI, MobilePhone

Example: May need to adjust for your local env
{
  "firstName": "Leo",
  "middleInitials": "J",
  "lastName": "Spaceman",
  "email": "user5435@userdomain.com",
  "telephone": "212-555-5555",
  "roles": [
     {
      "name": "Transportation Invoicing Officer",
      "roleType": "transportation_invoicing_officer"
    }
  ],
  "transportationOfficeId": "f344dde8-1c04-4a37-b66d-3f1bab7c494c",
  "edipi": "1111111110",
  "otherUniqueId": "1111111110",
  "rejectionReason": "string",
  "status": "APPROVED"
}

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

**Review by traskowskycaci — APPROVED** [review]
Commits match the int branch. Might need a spectral count increase here so pre_test is happy

**Comment by cameroncaci** [comment]
18825 should be going through today

**Comment by TevinAdams** [comment]
> 18825 should be going through today

This is ready to go.

File: swagger/admin.yaml [hunk 1]
```diff
@@ -43,10 +43,6 @@ tags:
     description: Information about notifications
     externalDocs:
       url: https://transcom.github.io/mymove-docs/docs/api
-  - name: Requested office users
-    description: Information about Requested Office UI users
-    externalDocs:
-      url: https://transcom.github.io/mymove-docs/docs/api
```
Comment (danieljordan-caci) [inline swagger/admin.yaml:49]: We don't want this removed

File: swagger/admin.yaml [hunk 2]
```diff
@@ -1256,7 +1270,7 @@ paths:
       produces:
         - application/json
       summary: Get a Requested Office User
-      description: Retrieving a single office user in any status. This endpoint is used in the Admin UI that will allow the admin user to view the user's relevant data.
```
Comment (danieljordan-caci) [inline swagger/admin.yaml:1259]: Also don't want this

Comment (TevinAdams) [reply swagger/admin.yaml:1259]: @danieljordan-caci This file is auto generated. On main the swagger-def/admin.yaml has "Retrieving a single office user in any status". The file also has no text matching  "Retrieving a single office user in any status. This endpoint is used in the Admin UI that will allow the admin user to view the user's relevant data.". Not sure how this happened but doesn't seem right.

**Comment by TevinAdams** [comment]
Removing and creating new branch. This current one seems pretty messed up but I think something went wrong between solving merge conflicts with things that were reverted.

**Comment by danieljordan-caci** [comment]
> Removing and creating new branch. This current one seems pretty messed up but I think something went wrong between solving merge conflicts with things that were reverted.

Probably a good call. Just cherry pick your commits and should be good to go. I think all the code you need is in `integrationTesting` already

**Pull request closed without merging**